src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -145,22 +145,43 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     // processor needed for BBCode to HTML translation
     TextProcessor processor = BBProcessorFactory.getInstance().createFromResource("kefirbb.xml");
 
-    String messageContent = request.getParameter("message");
+    String messageContent;
+    if(request.getParameter("message") != null)
+        messageContent = request.getParameter("message");
+    else
+        messageContent = request.getParameter("reply");
 
     // this removes any HTML from the message content and then parses any BBCode to HTML
     String cleanedMessageContent = processor.process(Jsoup.clean(messageContent, Whitelist.none()));
 
     String cleanedAndEmojiCheckedMsg = EmojiParser.parseToUnicode(cleanedMessageContent);
 
-    Message message =
+   
+    if(request.getParameter("reply")!= null) {
+        String mainMessageUUID = request.getParameter("messageUUID");
```

Comment:
WDYT about using "parentMessageId" instead to make it a little clearer what this parameter holds.